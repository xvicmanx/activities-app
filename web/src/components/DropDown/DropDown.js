import React, { useState } from 'react';
import { Dropdown as BulmaDropdown } from 'react-bulma-components';
import './DropDown.css';

const DropDown = ({
  name,
  value,
  onChange,
  items,
  placeholder
}) => {
  let options = [];

  if (items !== undefined) {
    options = items;
  }

  const handleChange = val => {
    onChange({
      persist: () => {},
      target: {
        name,
        value: val,
      },
    });
  };

  return (
    <BulmaDropdown
      value={value}
      onChange={handleChange}
    >
      <BulmaDropdown.Item>
        {placeholder}
      </BulmaDropdown.Item>
      {options.map((el, i) => {
        return (
          <BulmaDropdown.Item value={i} key={el.id}>
            {el.name}
          </BulmaDropdown.Item>
        );
      })}
    </BulmaDropdown>
  );
};

export default DropDown;
