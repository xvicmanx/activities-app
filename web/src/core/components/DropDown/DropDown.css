.dropdown-content {
  max-height: 200px;
  overflow: auto;
}
