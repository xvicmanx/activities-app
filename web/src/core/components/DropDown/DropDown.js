import React from 'react';
import { Dropdown as BulmaDropdown } from 'react-bulma-components';
import './DropDown.css';

const DropDown = ({
  name,
  value,
  onChange,
  items,
  placeholder
}) => {
  const handleChange = val => {
    onChange({
      persist: () => {},
      target: {
        name,
        value: val,
      },
    });
  };

  return (
    <BulmaDropdown
      value={value}
      onChange={handleChange}
    >
      <BulmaDropdown.Item>
        {placeholder}
      </BulmaDropdown.Item>
      {(items || []).map((el) => {
        return (
          <BulmaDropdown.Item value={el.id} key={el.id}>
            {el.name}
          </BulmaDropdown.Item>
        );
      })}
    </BulmaDropdown>
  );
};

export default DropDown;
