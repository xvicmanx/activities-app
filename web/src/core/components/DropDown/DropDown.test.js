import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DropDown from './DropDown';

Enzyme.configure({ adapter: new Adapter() })

describe('DropDown', () => {
  const items = [
    {
      id: 1,
      name: 'first',
    },
    {
      id: 2,
      name: 'second',
    },
  ];

  const props = {
    name: 'test-drop',
    value: 1,
    items,
    placeholder: 'Please select',
  };

  it('renders properly', () => {
    const component = renderer.create(<DropDown
      {...props}
      onChange={() => {}}
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls the onChange handler when dropdown value changes', () => {
    const onChangeMock = jest.fn();
    const event = {
      target: { name: 'test-drop', value: 2 },
    };
    const component = Enzyme.mount(<DropDown {...props} onChange={onChangeMock} />);
    component.find('.dropdown-item').at(4).simulate('click');
    expect(onChangeMock).toBeCalledWith(expect.objectContaining({
      target: event.target,
    }));
  });
});
