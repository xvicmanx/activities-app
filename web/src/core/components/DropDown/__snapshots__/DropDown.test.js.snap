// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown renders properly 1`] = `
<div
  className="dropdown"
>
  <div
    className="dropdown-trigger"
    onClick={[Function]}
    role="presentation"
  >
    <button
      className="button"
      disabled={false}
      onClick={[Function]}
      tabIndex={0}
    >
      <span>
        first
      </span>
      <span
        className="icon is-small"
      >
        <i
          className="rbc rbc-angle-down"
        />
      </span>
    </button>
  </div>
  <div
    className="dropdown-menu"
    id="dropdown-menu"
    role="menu"
  >
    <div
      className="dropdown-content"
    >
      <div
        className="dropdown-item"
        onClick={[Function]}
        role="presentation"
        title={0}
      >
        Please select
      </div>
      <div
        className="dropdown-item is-active"
        onClick={[Function]}
        role="presentation"
        title={1}
      >
        first
      </div>
      <div
        className="dropdown-item"
        onClick={[Function]}
        role="presentation"
        title={2}
      >
        second
      </div>
    </div>
  </div>
</div>
`;
